### ISSUE TITLE
Hashable instance for Fingerprint

### ISSUE THREAD

**Issue by andrewthad** [tibbe/hashable#156]
We've already got a Hashable instance for TypeRep. Having one for Fingerprint would be handy too.

**Comment by treeowl** [comment]
Agreed. Note that the instance for `TypeRep` doesn't actually salt properly:

```haskell
forall (s1 :: Salt) (s2 :: Salt).
hashWithSalt s1 = hashWithSalt s2
```

I suppose the logic here is that there should be few enough collisions for `HashMap` purposes using just one of the words of the `Fingerprint`?

**Comment by treeowl** [comment]
Actually.... that logic applies to `TypeRep` fingerprints, but I'm not sure it applies to fingerprints in general. Are the two words sometimes independent?

**Comment by hvr** [comment]
Resolved by #157
